### PR TITLE
Fixes #10300 - adding new image power API support

### PR DESCRIPTION
**Pull request by lzap**
New image power api with kexec support. This is
for https://github.com/theforeman/foreman-discovery-image/pull/21.

Tests fail because of http://projects.theforeman.org/issues/11708

### CODE REVIEW

**Comment by lzap** [comment]
Once https://github.com/theforeman/foreman_discovery/pull/214 is merged we can re-run tests.

File: app/services/foreman_discovery/node_api/power.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      end
+    end
+  end
+
+  class PowerService < NodeResource
+    def url
+      @args[:url] + "/power"
+    end
+
+    def reboot
```
Comment (ohadlevy) [inline app/services/foreman_discovery/node_api/power.rb:23]: maybe kexec should be a flag on reboot?

File: app/services/foreman_discovery/node_api/power.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    end
+
+    def kexec(json)
+      put(json, "/kexec")
+    end
+
+  end
+
+  # legacy /reboot call direct
+  class PowerLegacyDirectService < NodeResource
```
Comment (ohadlevy) [inline app/services/foreman_discovery/node_api/power.rb:34]: add depreciation warnings using Foreman::Deprecate ?

File: app/services/foreman_discovery/node_api/power.rb (hunk 1 above)
Comment (lzap) [inline app/services/foreman_discovery/node_api/power.rb:23]: Nah its a complete different operation with its own input. And also, this gave us chance to get rid of reusing BMC interface once and forever.

File: app/services/foreman_discovery/node_api/power.rb (hunk 2 above)
Comment (lzap) [inline app/services/foreman_discovery/node_api/power.rb:34]: Good idea, we still use this API to reboot legacy nodes.

**Comment by lzap** [comment]
[test]

**Comment by lzap** [comment]
@stbenjam tool a bit longer then I expected, thanks

**Comment by stbenjam** [comment]
Merged as b713c12a2546173ed5de95e47303ab99e5b2f19c, sorry for the delay @lzap

**Comment by lzap** [comment]
Thank you so much! :-) Yay.

**Pull request closed without merging**